From cf15d82cdc4a040d549e8a5598c70c8e7c334ecc Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Wed, 24 Apr 2013 09:38:42 +0300
Subject: [PORT FROM MAIN] Telephony: Query the call status for failure cases

BZ: 27617

Android framework queries the call status upon
completion of each call operation or upon the receival
of RIL_UNSOL_RESPONSE_CALL_STATE_CHANGED event.  Due to
this, on some use cases, wrong call UI is shown to the user.

When any of the call related requests fails, trigger
the call status query. This is achieved by setting
the needsPoll to true.

Change-Id: Ief263045c62b620c7feb975b47012af5502e7296
Orig-Change-Id: Ib9708f36beb58c40ff79e077f02e20edae22b69c
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../internal/telephony/gsm/GsmCallTracker.java     |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/java/com/android/internal/telephony/gsm/GsmCallTracker.java b/src/java/com/android/internal/telephony/gsm/GsmCallTracker.java
index f866078..3a7dfa1 100644
--- a/src/java/com/android/internal/telephony/gsm/GsmCallTracker.java
+++ b/src/java/com/android/internal/telephony/gsm/GsmCallTracker.java
@@ -861,6 +861,9 @@ public final class GsmCallTracker extends CallTracker {
 
             case EVENT_OPERATION_COMPLETE:
                 ar = (AsyncResult)msg.obj;
+                if (ar.exception != null) {
+                    mNeedsPoll = true;
+                }
                 operationComplete();
             break;
 
@@ -871,6 +874,7 @@ public final class GsmCallTracker extends CallTracker {
                 ar = (AsyncResult)msg.obj;
                 if (ar.exception != null) {
                     mPhone.notifySuppServiceFailed(getFailedService(msg.what));
+                    mNeedsPoll = true;
                 }
                 operationComplete();
             break;
-- 
1.7.4.1

